packages/hash/api/src/model/ontology/dataType.model.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  DataType as DataTypeSchema,
+  GraphApi,
+} from "@hashintel/hash-graph-client";
+
+import { DataType as DataTypeModel } from "../index";
```

Comment:
I'm slightly confused, isn't this importing something from the file above, which in turn imports it from this file?

Reply:
The import isn't required, as we could use `__DataType` instead. But yes, it would import from the file about and then use the same exact class defined below

Reply:
Yeah I think the only reason we went with this is because `__ClassName` is a bit more awkward to use and see than `ClassName`

Reply:
Is this an established pattern? Does anywhere else do this? Because this is genuinely one of the most confusing things I've seen in a while 🙃

Reply:
What's the reason for the `__` prefix anyway?

Reply:
It was a hacky solution at the time, Ahmad and I will try to find an alternative now

Reply:
Ahmad came up with a way for us to avoid having to use `__`: [don't define class names in the file in which they are defined](https://github.com/hashintel/hash/pull/894/commits/bebeae6bc0e2050b6251db7dce51b1b91875ad3a)

Reply:
How does this affect IDE functionalities like "go to declaration" and the equivalent? Maybe the underscores are the lesser evil if this is our only other idea :(

Reply:
It goes to the exact class definition and not the import, so should behave as expected (the same way `__` did)

Reply:
Great!

Reply:
`cmd + click` on usages of `PropertyTypeModel` still go to the class declaration 👍

Reply:
(at least in VSCode)

Reply:
(I mean I still don't know how I feel about this solution either but I think we're just fighting TypeScript's abilities here and JS not really letting you have things as being private within scopes 🙃 )

Reply:
Yeah, visibility modifiers would be nice here but unfortunately don't exist in JS/TS to my knowledge. The only solution I can think of is to declare the model classes in their own package, so that we can control what is exposed - but I don't think that's worth the effort

Reply:
TypeScript's own documentation recommends https://api-extractor.com/ for proper scoping/visibility (source https://www.typescriptlang.org/tsconfig#stripInternal). Kind of insane we have to pull in a third party dep for that